### DISCUSSION TITLE
How to match attribute_name with prefix in HTML.

### DISCUSSION THREAD

**Discussion by alexzhang1030** [ast-grep/ast-grep#1129] [General]
Here is a [playground](https://ast-grep.github.io/playground.html#eyJtb2RlIjoiQ29uZmlnIiwibGFuZyI6Imh0bWwiLCJxdWVyeSI6IjwkQSAkJCQgPiQkJDwvJEE+IiwicmV3cml0ZSI6IiIsImNvbmZpZyI6InJ1bGU6XG4gIGtpbmQ6IGF0dHJpYnV0ZVxuICByZWdleDogXCI6Lio9XCIiLCJzb3VyY2UiOiI8ZGl2PlxuICA8YXJ0aWNsZSBjbGFzcz1cImZvb1wiPlxuICAgIExvcmVtIGlwc3VtLi4uLlxuICA8L2FydGljbGU+XG4gIDxwIGNsYXNzPVwiYmFyXCI+XG4gICAgTG9yZW0gaXBzdW0uLi4uXG4gIDwvcD5cbiAgPGRpdiBjbGFzcz1cImZvb1wiIDphPVwiYVwiIDpiPVwiYlwiIGM9XCJjXCIgOmQ+XG4gICAgTG9yZW0gaXBzdW0uLi4uXG4gIDwvZGl2PlxuICA8ZGl2IDpjPVwiY1wiPjwvZGl2PlxuPC9kaXY+XG4ifQ==)

I'm not really sure this is the best solution, and I don't know how to fix this from `:$A="$A"` to `:$A`

Cause `:a="a"` will be parsed to `attr_name: ':a', attr_value: 'a' `, can we add some options like `prefix` or `modifiers` to support `:$A="$A"`?

**Comment by HerringtonDarkholme** [comment]
```yaml
rule:
  kind: attribute
  regex: ":.*="
```

I think this is a good solution.

**Reply by alexzhang1030** [reply]
Then can we use capture group for fix/transform in regex？

**Reply by HerringtonDarkholme** [reply]
you can use it in [`replace`](https://ast-grep.github.io/reference/yaml/transformation.html#replace) transform. And then use the transformed meta var in `fix`
